src/client/app/utils/helperFunctions.js

```diff
@@ -1,13 +1,13 @@
 const debounce = (func, wait, immediate) => {
     let timeout;
 
-    return function () {
-        const args = arguments;
+    return function debouncify (...args) {
+        const params = args;
         const later = () => {
             timeout = null;
 
             if (!immediate) {
-                func.apply(this, args);
+                func.apply(this, params);
```

Comment:
Not sure about using this here. Maybe the caller should take care of binding func if they really care.